scripts/screeps_tencent_batch_rl_runner.py

```
@@ -918,6 +971,411 @@ def remote_training_report_path(artifact_dir: Path, run_id: str) -> Path:
+
+def extract_completed_run_ticks(summary: dict[str, Any], reports: Sequence[dict[str, Any]]) -> int | None:
+    raw = (
+        path_value(summary, "inputs", "ticks")
+        or first_value(reports, (("simulation", "ticks"), ("source", "initialConditions", "ticks")))
+    )
+    value = numeric_value(raw)
+    return int(value) if value is not None and value == int(value) else None
+
+
+def extract_dead_tier_metric(reports: Sequence[dict[str, Any]]) -> dict[str, Any] | None:
+    for field in ("variantResults", "ranking"):
```

Ruff (0.15.12)

[error] Import `field` from line 30 shadowed by loop variable

(F402)